test/replication-luatest/gh_8675_applier_pause_assert_test.lua

```diff
@@ -0,0 +1,58 @@
+local t = require('luatest')
+local server = require('luatest.server')
+local replica_set = require('luatest.replica_set')
+
+local g = t.group('gh-8675-applier-pause')
+
+local fio = require('fio')
+
+local set_log_before_cfg = [[
+    local logfile = require('fio').pathjoin(
+        os.getenv('TARANTOOL_WORKDIR'),
+        os.getenv('TARANTOOL_ALIAS') .. '.log'
+    )
+    require('log').cfg{log = logfile}
+]]
+
+g.before_each(function(cg)
+    cg.replica_set = replica_set:new{}
+    cg.server = cg.replica_set:build_and_add_server{
+        alias = 'server',
+        box_cfg = {
+            replication_connect_timeout = 0.1,
+            replication_timeout = 0.01,
+        },
+        env = {['TARANTOOL_RUN_BEFORE_BOX_CFG'] = set_log_before_cfg},
+    }
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+local function assert_normal_exit(server)
+    local logfile = fio.pathjoin(server.workdir, server.alias .. '.log')
+    local msg = "can't initialize storage: Incorrect value for option " ..
+                "'replication': failed to connect to one or more replicas"
+    t.helpers.retrying({}, function()
+        t.assert(server:grep_log(msg, nil, {filename = logfile}),
+                 'Configuration is failed')
+    end)
+    t.assert_equals(server:grep_log('Assertion failed', nil,
+                                    {filename = logfile}),
+                    nil, 'Exit without a crash')
+end
+
+g.test_crash_on_reconfiguration_failure = function(cg)
+    cg.server.box_cfg.replication = {
+        server.build_listen_uri('nonexistent', cg.replica_set.id),
+    }
+    cg.server:start{wait_until_ready = false}
+    assert_normal_exit(cg.server)
+    cg.server.box_cfg.replication = {
```

Comment:
These tests seems independent from each other and test different assertions. Why do we place it in one function? I suppose, we should apply the following diff here:

```diff
diff --git a/test/replication-luatest/gh_8675_applier_pause_assert_test.lua b/test/replication-luatest/gh_8675_applier_pause_assert_test.lua
index 0daaf06a4..a42e9f793 100644
--- a/test/replication-luatest/gh_8675_applier_pause_assert_test.lua
+++ b/test/replication-luatest/gh_8675_applier_pause_assert_test.lua
@@ -14,7 +14,7 @@ local set_log_before_cfg = [[
     require('log').cfg{log = logfile}
 ]]
 
-g.before_each(function(cg)
+g.before_all(function(cg)
     cg.replica_set = replica_set:new{}
     cg.server = cg.replica_set:build_and_add_server{
         alias = 'server',
@@ -26,7 +26,7 @@ g.before_each(function(cg)
     }
 end)
 
-g.after_each(function(cg)
+g.after_all(function(cg)
     cg.replica_set:drop()
 end)
 
@@ -49,6 +49,9 @@ g.test_crash_on_reconfiguration_failure = function(cg)
     }
     cg.server:start{wait_until_ready = false}
     assert_normal_exit(cg.server)
+end
+
+g.test_other = function(cg)
     cg.server.box_cfg.replication = {
         server.build_listen_uri(cg.server.alias, cg.replica_set.id),
         server.build_listen_uri('nonexistent', cg.replica_set.id),

```

This way the test won't stop on failing assertion `fiber() == applier->fiber` and will test `!applier->is_paused` too, if smth goes wrong.

Reply:
Thanks for paying attention!
Applied the following diff. I didn't touch the `before/after_each`, because if something goes wrong the server might not stop after the first test.
```diff
diff --git a/test/replication-luatest/gh_8675_applier_pause_assert_test.lua b/test/replication-luatest/gh_8675_applier_pause_assert_test.lua
index 0daaf06a4..f07da4a58 100644
--- a/test/replication-luatest/gh_8675_applier_pause_assert_test.lua
+++ b/test/replication-luatest/gh_8675_applier_pause_assert_test.lua
@@ -43,12 +43,15 @@ local function assert_normal_exit(server)
                     nil, 'Exit without a crash')
 end
 
-g.test_crash_on_reconfiguration_failure = function(cg)
+g.test_crash_without_self = function(cg)
     cg.server.box_cfg.replication = {
         server.build_listen_uri('nonexistent', cg.replica_set.id),
     }
     cg.server:start{wait_until_ready = false}
     assert_normal_exit(cg.server)
+end
+
+g.test_crash_with_self = function(cg)
     cg.server.box_cfg.replication = {
         server.build_listen_uri(cg.server.alias, cg.replica_set.id),
         server.build_listen_uri('nonexistent', cg.replica_set.id),
```